packages/react-components/react-nav-preview/src/components/NavItem/useNavItem.ts

```diff
@@ -13,18 +14,41 @@ import type { NavItemProps, NavItemState } from './NavItem.types';
  * @param props - props from this instance of NavItem
  * @param ref - reference to root HTMLAnchorElement of NavItem
  */
-export const useNavItem_unstable = (props: NavItemProps, ref: React.Ref<HTMLAnchorElement>): NavItemState => {
-  const { onClick, value, icon } = props;
+export const useNavItem_unstable = (
+  props: NavItemProps,
+  ref: React.Ref<HTMLButtonElement | HTMLAnchorElement>,
+): NavItemState => {
+  const { onClick, value, icon, as, href } = props;
 
   const { selectedValue, onRegister, onUnregister, onSelect } = useNavContext_unstable();
 
+  const rootElementType = as || (href ? 'a' : 'button');
+
   const selected = selectedValue === value;
 
   const innerRef = React.useRef<HTMLElement>(null);
-  const onNavItemClick = useEventCallback(
-    mergeCallbacks(onClick, event => onSelect(event, { type: 'click', event, value })),
+
+  const onNavItemClick: ARIAButtonSlotProps<'a'>['onClick'] = useEventCallback(event => {
+    onClick?.(event);
+
+    if (!event.defaultPrevented && isHTMLElement(event.target)) {
+      onSelect(event, { type: 'click', event, value });
+    }
+  });
+
+  const _navItem = slot.always<ARIAButtonSlotProps<'a'>>(
```

Comment:
any reason not to rename this directly to root?
```suggestion
  const root = slot.always<ARIAButtonSlotProps<'a'>>(
```